src/__test__/client.test.ts

```diff
@@ -0,0 +1,213 @@
+} from '../errors'
+
+let client: Client
+
+const auth = {
+  username: 'testuser',
+  password: 'lq6zequb-gYTdF2_ZEUtr8ywTXzLYtknzWU4nV8uVoo=',
+}
+const url = 'http://localhost:18443'
+
+describe('client test', () => {
+  beforeAll(() => {
+    client = new Client({
+      baseURL: url,
+      auth,
+    })
+  })
+  test('getnewaddress', async () => {
+    const address = await client.getNewAddress(undefined, 'bech32')
+    expect(address.startsWith('bcrt')).toBeTruthy()
+  })
+  test('getbalance', async () => {
+    const address = await client.getNewAddress()
+    await client.generateToAddress(101, address)
+    const balance = await client.getBalance()
+    expect(balance).toEqual(50)
+  })
+  test('lockutxos', async () => {
+    const address = await client.getNewAddress()
+    await client.generateToAddress(101, address)
+    const utxos = await client.listUnspent()
+    const toLock = { txid: utxos[0].txid, vout: utxos[0].vout }
+    await client.lockUnspent(false, [toLock])
+    let lockList = await client.listLockUnspent()
+    expect(lockList).toContainEqual(toLock)
+
+    await client.lockUnspent(true)
+
+    lockList = await client.listLockUnspent()
+    expect(lockList.length).toEqual(0)
+  })
+  test('importaddress', async () => {
+    const address = 'bcrt1qlgmznucxpdkp5k3ktsct7eh6qrc4tju7ktjukn'
+    await client.importAddress(address)
+    const info = await client.getAddressInfo(address)
+    expect(info.iswatchonly).toBeTruthy()
+  })
+  test('importpublickey', async () => {
+    const privkey =
+      '0000000000000000000000000000000000000000000000000000000000000001'
+    const pubkey = cfdjs.GetPubkeyFromPrivkey({ privkey, isCompressed: true })
+      .pubkey
+    const address = cfdjs.CreateAddress({
+      keyData: { type: 'pubkey', hex: pubkey },
+      network: 'regtest',
+      hashType: 'p2wpkh',
+    }).address
+    await client.importPubKey(pubkey)
+    const info = await client.getAddressInfo(address)
+    expect(info.iswatchonly).toBeTruthy()
+  })
+  test('createWallet', async () => {
+    const walletParams = [
+      { name: 'alice' },
+      { name: 'bob', disablePrivateKeys: true },
+      { name: 'carol', disablePrivateKeys: false, blank: true },
+      {
+        name: 'denise',
+        disablePrivateKeys: false,
+        blank: false,
+        passphrase: 'pass',
+      },
+      {
+        name: 'emily',
+        disablePrivateKeys: false,
+        blank: false,
+        passphrase: '',
+        avoid_reuse: true,
+      },
+    ]
+
+    for (const params of walletParams) {
+      const result = await client.createWallet(
+        params.name,
+        params.disablePrivateKeys,
+        params.blank,
+        params.passphrase,
+        params.avoid_reuse
+      )
+
+      expect(result.name).toEqual(params.name)
+      const expectedWarning =
+        params.passphrase === undefined || params.passphrase === ''
+          ? 'Empty string given as passphrase, wallet will not be encrypted.'
```

Comment:
maybe move this warning string into a const as it seems to come from `bitcoind`

Reply:
Used only here would rather keep it like this.

Reply:
understood